openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ReturnTypeGenerator.java

```diff
@@ -174,8 +174,10 @@ public ReturnTypeDescriptorNode getReturnTypeDescriptorNode(Map.Entry<PathItem.H
             String typeDescriptionTemplate = "#            %s (%s)%n";
             for (String description : returnDescriptions) {
                 String[] values = description.split("\\|");
+                // Replace new lines and tabs in the description by space.
+                String responseDescription = values[1].replaceAll("[\\r\\n\\t]", " ");
```

Comment:
Can't we support new lines and generate multiline descriptions in such cases? appending to a single line might not be the best idea when it comes to long descriptions and intentionally kept multi lines (e.g. code snippets) IMO.

Reply:
Here we are using an template to document the return types. The issue we are trying to address here happens when the function return is a union type. 

The template we are using to format the documentation in such cases is as below

```bal
    # + return - returns can be any of following types
    # <Type1> (<Response Description>)
    # <Type2> (<Response Description>)
```

Ex: 

```bal
    # + return - returns can be any of following types
    # Pets (OK. An paged array of pets)
    # http:Response (unexpected error)
```

The description of a particular return type is inside braces. This design was discussed [here](https://docs.google.com/document/d/1Jd0KSkd0fW9mWqk7tKIIX-ds9n42FLmIYK3HN8gssGE/edit?disco=AAAAqsRE6Oc)

According to the yaml of this description, if we support new lines, the generation would be as below

```bal
    # + return - returns can be any of following types
    # Pets (OK. 
    # An paged array of pets)
    # http:Response (unexpected error)
```

Since this a braced statement, having it in the same line seems more suitable IMO.

Reply:
@aneeshafedo I see. Yeah it makes sense. +1 for the approach :)